README.md

```diff
@@ -260,38 +260,39 @@ The Trillian service is multi-tenanted &ndash; a single Trillian installation
 can support multiple Merkle trees in parallel, distinguished by their `TreeId`
 &ndash; and each tree operates in one of two modes:
 
- - **Log** mode: an append-only collection of items.
+ - **Log** mode: an append-only collection of items; this has two sub-modes:
+   - normal Log mode, where the Trillian service assigns sequence numbers to
+     new tree entries as they arrive
+   - 'preordered' Log mode, where the unique sequence number for entries in
+     the Merkle tree is externally specified
  - **Map** mode: a collection of key:value pairs.
 
 In either case, Trillian's key transparency property is that cryptographic
 proofs of inclusion/consistency are available for data items added to the
 service.
 
-### Personalities
 
-The Trillian service expects to be paired with additional code that is specific
-to the particular application of the transparent store; this is known as a
-*personality*.
+### Personalities
 
-The primary purpose of a personality is to implement **admission criteria** for
-the store, so that only particular types of data are added to the store. For
-example, a Certificate Transparency log only accepts data items that are valid
-certificates; a "CT Log" personality would police this, so that the Trillian
-service can process all incoming data blindly.
+To build a complete transparent application, the Trillian core service needs
+to be paired with additional code, known as a *personality*, that provides
+functionality that is specific to the particular application.
 
-A personality may also perform **canonicalization** on incoming data, to
-convert equivalent formulations of the same underlying data to a single
-canonical format, avoiding needless duplication.  (For example, keys in
-JSON dictionaries could be sorted, or Unicode string data could be normalised.)
+In particular, the personality is responsible for:
 
-The per-application personality is also responsible for providing an
-externally-visible interface, typically over HTTP[S].
+ * **Admission Criteria** &ndash; ensuring that submissions comply with the
+   overall purpose of the application.
+ * **Canonicalization** &ndash; ensuring that equivalent versions of the same
```

Comment:
Maybe clarify that this is limited to providing a canonical identifier for the data such that equivalent items receive can be identified and deduplicated by Trillian (currently it could be interpreted to mean that the personality must do the de-dupe).